### PR TITLE
[node] Implement JSONAPI operations

### PR DESCRIPTION
**Pull request by joelalejandro**
This is a draft implementation of what's being discussed in #1250.

### CODE REVIEW

File: packages/node/src/api-router.ts [hunk 1]
```diff
@@ -32,6 +32,41 @@ import {
...
+  UninstallController: { type: "app", op: "uninstall" },
+  UninstallVirtualController: { type: "app", op: "uninstallVirtual" },
+  TakeActionController: { type: "app", op: "takeAction" },
+  UpdateStateController: { type: "app", op: "updateState" },
+  GetAppInstanceController: { type: "app", op: "get", params: ["id"] },
+  GetAppInstanceStateController: { type: "app", op: "getState" },
+  GetInstalledAppInstancesController: { type: "app", op: "get" }
+};
+
 const controllers = [
```
Comment (ebryn) [inline packages/node/src/api-router.ts]: `const controllers = Object.keys(controllersToOperations);`

Comment (joelalejandro) [reply packages/node/src/api-router.ts]: Not sure about that. Notice that here we store the Controller constructors...

File: packages/node/src/api-router.ts [hunk 2]
```diff
@@ -31,6 +31,41 @@ import {
 } from "./methods";
 import { NODE_EVENTS } from "./types";
 
+// Maps controllers to JSONAPI operations.
+export const controllersToOperations = {
```
Comment (snario) [inline packages/node/src/api-router.ts]: Why are some ops in a fixed set like `get` or `add` but others are strings like `getFreeBalanceState`?

Comment (snario) [reply packages/node/src/api-router.ts]: Is there an intro to `@ebryn/jsonapi-ts` with motivation?

File: package.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
     "lint": "lerna run lint --parallel --no-bail",
     "lint:fix": "lerna run lint:fix --parallel --no-bail",
     "run:playground": "lerna run start --parallel --scope=**/playground --scope=**/playground-server --scope=**/dapp-high-roller --scope=**/dapp-tic-tac-toe",
+    "run:playground:server": "lerna run start --parallel --scope=**/playground-server",
```
Comment (cf19drofxots) [inline package.json]: Could simply do `yarn start` from the playground-server package, no?

File: packages/node/src/events/propose-install-virtual/controller.ts [hunk 4]
```diff
@@ -34,10 +34,24 @@ export default async function proposeInstallVirtualEventController(
...
   }
 
-  const relayedProposalMsg: ProposeVirtualMessage = {
-    from: requestHandler.publicIdentifier,
-    type: NODE_EVENTS.PROPOSE_INSTALL_VIRTUAL,
-    data: nodeMsg.data
+  const relayedProposalMsg: NodeOperation = {
+    meta: {
+      from: requestHandler.publicIdentifier,
+      requestId: ""
```
Comment (cf19drofxots) [inline packages/node/src/events/propose-install-virtual/controller.ts]: Probably outside the scope of this PR but we should discuss needing `requestId` - from what I've seen so far it's been empty / not referenced at all.

**Comment by joelalejandro** [comment]
Cancelled in favour of JSONRPC.

**Pull request closed without merging**